instrumentation/jboss-logmanager/jboss-logmanager-mdc-1.1/javaagent/build.gradle.kts

```diff
@@ -12,5 +12,5 @@ muzzle {
 }
 
 dependencies {
-  library("org.jboss.logmanager:jboss-logmanager:1.1.0.GA")
+  library("org.jboss.logmanager:jboss-logmanager:1.3.0.Final")
```

Comment:
If possible we compile and test against the earliest version supported. If introducing a new instrumentation feature makes the instrumentation incompatible then the version number should be bumped everywhere, in module name, package name, docs etc.
As far as I understand you are only calling `getMdcCopy` from test, which is using groovy, so you can probably change this back to 1.1.0. In test you could check whether this method is present before calling it. Something like https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/efd40f8cdf73f33b8a055359d7fe8bed5afaf54c/instrumentation/netty/netty-3.8/javaagent/src/test/groovy/Netty38ClientTest.groovy#L33 When you run tests with `-PtestLatestDeps=true` tests will run against latest version of `jboss-logmanager` and thus test your changes too. Alternatively it is possible to define `testSets` using `org.unbroken-dome.test-sets` plugin so you can test with multiple versions each of which can have its own tests, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jsf/jsf-mojarra-1.2/javaagent/build.gradle.kts Getting test sets to work exactly as you want can be tricky.

Reply:
Thank for the advice, I re-wrote the test, and also reverted the dependency version. 

I found out the hard way that I must not use an `if` statement within the `then` block. I have to admit I've never used Spock to write tests, so I'm happy to hear a better way to do this.

Reply:
Try
```groovy
if (hasGetMdcCopy) {
  assert logRecords.get(2).getMdcCopy().get("trace_id") == span1.spanContext.traceId
  assert logRecords.get(2).getMdcCopy().get("span_id") == span1.spanContext.spanId
  assert logRecords.get(2).getMdcCopy().get("trace_flags") == "01"
}
```

Reply:
Thanks for pointing me to the `assert` keyword that seems to be necessary within an if block. Without that it evaluated always to true. Ready for a (final?) review.